src/main/java/io/github/cottonmc/libcd/api/AdvancementRewardsManager.java

```diff
@@ -0,0 +1,26 @@
+package io.github.cottonmc.libcd.api;
+
+import com.google.common.collect.Maps;
+import com.google.gson.JsonDeserializer;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.util.Identifier;
+
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+public class AdvancementRewardsManager {
```

Comment:
To fit with the other systems in LibCD, this shouldn't be set up as static. It should have an `INSTANCE`, and we should probably add a setup command for it to `LibCDInitializer`.